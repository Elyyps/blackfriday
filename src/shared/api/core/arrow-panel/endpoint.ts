import { generateArrowPanelData } from "./generate-dummy-data";

export const generateTabsDummyData = generateArrowPanelData();
