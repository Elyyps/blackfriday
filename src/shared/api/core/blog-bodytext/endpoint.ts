import { generateDummyBlogBodyText, generateBlackFriday2019Text } from "./generate-dummy-data";

export const generateBlogBodyTextData = generateDummyBlogBodyText;
export const generateBlackFriday2019TextData = generateBlackFriday2019Text;
