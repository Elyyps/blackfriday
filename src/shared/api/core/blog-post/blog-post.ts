export class BlogPost {
  public content: string;
  public image: string;
  public link: string;
  public linkTitle: string;
  public title: string;
}
