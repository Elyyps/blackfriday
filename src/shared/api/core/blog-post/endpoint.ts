import { generateDummyBlogPost, generate2DummyBlogPosts } from "./generate-dummy-data";

export const generateBlogPostData = generateDummyBlogPost;
export const generate2BlogPostsData = generate2DummyBlogPosts;
