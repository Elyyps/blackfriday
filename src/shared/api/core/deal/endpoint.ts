import { Deal } from "./deal";
import { generateDemoDealDummyData } from "./generate-dummy-data";
import { FilterItem } from "../filter/filter-item";
import { shuffle } from "@app/util/array";
/* tslint:disable */
export interface IDealResult {
  deals: Deal[];
  totalResults: number;
}

const initialStoresResult = [...generateDemoDealDummyData()];

const getStores = (
  skip: number,
  take: number,
  storeFilters: FilterItem[],
  categoryFilters: FilterItem[],
  brandFilters: FilterItem[],
  sortBy?: string
): IDealResult => {
  const selectedCategoryFilters = getSelectedFilters(categoryFilters);
  const selectedBrandFilters = getSelectedFilters(brandFilters);
  const selectedStoreFilters = getSelectedFilters(storeFilters);

  let result = [...initialStoresResult];

  if (selectedCategoryFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.categories.filter(category => {
        return selectedCategoryFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedBrandFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedBrandFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedStoreFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedStoreFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  const totalResults = result.length;

  result = result.splice(skip, take);

  switch (sortBy) {
    case "Relevantie":
      // Keep default sorting
      break;
    case "Nieuwste":
      result.reverse();
      break;
    case "Populair":
      result = shuffle(result);
      break;
    case "Winkels A - Z":
      result = result.sort((a, b) => {
        if (a.name < b.name) {
          return -1;
        }
        if (b.name < a.name) {
          return 1;
        }

        return 0;
      });
      break;
    case "Winkels Z - A":
      result = result.sort((a, b) => {
        if (a.name > b.name) {
          return -1;
        }
        if (b.name > a.name) {
          return 1;
        }

        return 0;
      });
      break;
    default:
      /* tslint:disable:switch-final-break  */
      // keep default sorting
      break;
  }

  return {
    deals: result,
    totalResults
  };
};

const getSelectedFilters = (filterItems: FilterItem[]): string[] | undefined => {
  const selectedFilterItems = filterItems.filter(item => item.isSelected);

  if (selectedFilterItems && selectedFilterItems.length > 0) {
    return selectedFilterItems.map(item => item.displayName);
  }

  return undefined;
};

const getAmountForCategory = (category: string): number =>
  initialStoresResult.filter(deal => deal.categories.indexOf(category) > -1).length;

const getAmountForBrand = (brand: string): number =>
  initialStoresResult.filter(deal => deal.availableBrands.indexOf(brand) > -1).length;

const getAmountForStore = (store: string): number =>
  initialStoresResult.filter(deal => deal.stores.indexOf(store) > -1).length;

export const DealApi = {
  getStores,
  initialStoresResult,
  getAmountForCategory,
  getAmountForBrand,
  getAmountForStore
};
