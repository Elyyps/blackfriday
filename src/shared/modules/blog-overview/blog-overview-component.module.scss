.blog-overview {
  &__holder {
    @media (min-width: $breakpoint-small) {
      display: flex;
      align-items: flex-start;
    }
  }
  &__content {
    flex: 1 1 auto;
    margin-bottom: 20px;
  }
  &__sidebar {
    @media (min-width: $breakpoint-small) {
      width: 40%;
      flex-shrink: 0;
      margin-left: 24px;
    }
    @media (min-width: $breakpoint-medium) {
      width: 31%;
      margin: 0 0 0 40px;
    }
  }
  .posts-list {
    > div {
      padding: 16px 0;
      border-bottom: 1px solid $tint-gray;
      @media (max-width: $breakpoint-small-max) {
        padding: 10px 0;
      }
      &:first-child {
        padding-top: 0;
      }
    }
  }
}
