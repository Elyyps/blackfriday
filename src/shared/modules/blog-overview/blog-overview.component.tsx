import * as React from "react";
import styles from "./blog-overview-component.module.scss";
import { BlogPostComponent } from "@app/core/blog-post";
import { BlogOverviewModule } from "@app/api/modules/blog-overview/blog-overview";
import { SidebarComponent } from "@app/core/sidebar";

export interface IBlogOverviewComponentProps {
  blog: BlogOverviewModule;
}

const BlogOverviewComponent = (props: IBlogOverviewComponentProps) => (
  <div className={styles["blog-overview"]}>
    <div className="uk-container">
      <div className={styles["blog-overview__holder"]}>
        <div className={styles["blog-overview__content"]}>
          <div className={styles["posts-list"]}>
            {props.blog.posts &&
              props.blog.posts.map((item, key) => (
                <div key={key}>
                  <BlogPostComponent posts={item} />
                </div>
              ))}
          </div>
        </div>

        <div className={styles["blog-overview__sidebar"]}>
          <SidebarComponent featuredItemsGroups={props.blog.sidebar.featuredItemsGroups} />
        </div>
      </div>
    </div>
  </div>
);

export { BlogOverviewComponent };
