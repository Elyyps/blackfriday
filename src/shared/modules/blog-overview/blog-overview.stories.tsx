import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BlogOverviewComponent } from "./blog-overview.component";
import { generateBlogOverviewDataBelgian } from "@app/api/modules/blog-overview/endpoint";
const blog = generateBlogOverviewDataBelgian();

storiesOf("BlogOverview", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BlogOverviewComponent blog={blog} />);
