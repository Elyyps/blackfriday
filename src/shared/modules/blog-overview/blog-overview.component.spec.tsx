import React from "react";
import { shallow } from "enzyme";
import { generateBlogPostData } from "@app/api/core/blog-post/endpoint";
import { BlogOverviewComponent } from "./blog-overview.component";
const posts = generateBlogPostData();
describe("[BlogOverview]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<BlogOverviewComponent posts={posts} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
