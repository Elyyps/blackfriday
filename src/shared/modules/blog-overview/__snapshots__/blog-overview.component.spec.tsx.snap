// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BlogOverview] should render component without crashing 1`] = `
<div
  className="blog-overview"
>
  <div
    className="uk-container"
  >
    <div
      className="blog-overview__holder"
    >
      <div
        className="blog-overview__content"
      >
        <div
          className="posts-list"
        >
          <div
            key="0"
          >
            <BlogPostComponent
              posts={
                Object {
                  "content": "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
                  "image": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png",
                  "link": Object {
                    "title": "Lees meer",
                    "url": "",
                  },
                  "title": "Vijf voorspellingen van Black Friday Nederland",
                }
              }
            />
          </div>
          <div
            key="1"
          >
            <BlogPostComponent
              posts={
                Object {
                  "content": "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
                  "image": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png",
                  "link": Object {
                    "title": "Lees meer",
                    "url": "",
                  },
                  "title": "Top Black Friday deals van vorig jaar",
                }
              }
            />
          </div>
          <div
            key="2"
          >
            <BlogPostComponent
              posts={
                Object {
                  "content": "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
                  "image": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png",
                  "link": Object {
                    "title": "Lees meer",
                    "url": "",
                  },
                  "title": "Mis geen enkele Black Friday deal!",
                }
              }
            />
          </div>
          <div
            key="3"
          >
            <BlogPostComponent
              posts={
                Object {
                  "content": "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
                  "image": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png",
                  "link": Object {
                    "title": "Lees meer",
                    "url": "",
                  },
                  "title": "Wacht met shoppen tot Black Friday!",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="blog-overview__sidebar"
      >
        <SidebarComponent
          banner=""
          items={
            Array [
              Object {
                "icon": "",
                "text": "",
                "url": "",
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</div>
`;
